sacred/config/custom_containers.py

```diff
@@ -226,6 +226,58 @@ def revelation(self):
         return set()
 
 
+class ReadOnlyContainer:
+    def _readonly(self, *args, **kwargs):
+        raise SacredError(
+            'This {} is read-only!'.format(self.__class__.__name__,),
+            filter_traceback='always'
+        )
+
+
+class ReadOnlyDict(dict, ReadOnlyContainer):
```

Comment:
I think it would be better to inherit from UserDict and UserList instead https://docs.python.org/3/library/collections.html#collections.UserDict